libmamba/include/mamba/core/transaction_context.hpp

```diff
@@ -39,17 +39,30 @@ namespace mamba
             const std::pair<std::string, std::string>& py_versions,
             const std::vector<MatchSpec>& requested_specs
         );
+
+        TransactionContext(
+            const fs::u8path& target_prefix,
+            const fs::u8path& relocate_prefix,
+            const std::pair<std::string, std::string>& py_versions,
+            const std::vector<MatchSpec>& requested_specs
+        );
         ~TransactionContext();
         bool try_pyc_compilation(const std::vector<fs::u8path>& py_files);
         void wait_for_pyc_compilation();
 
+        const fs::u8path & get_relocated_prefix()const;
+
         bool has_python;
         fs::u8path target_prefix;
         fs::u8path site_packages_path;
         fs::u8path python_path;
         std::string python_version;
         std::string old_python_version;
         std::string short_python_version;
+
+        fs::u8path relocate_prefix;
+        bool relocate = false;
```

Comment:
I think we can just always store two `u8path`. One that is the place where files are installed to, and one that is used to exchange prefix information from binarys. Most of the time they will just hbe the same.

Reply:
agreed